packages/nc-gui/composables/useKanbanViewData.ts

```diff
@@ -0,0 +1,432 @@
+import type { ComputedRef, Ref } from 'vue'
+import type { Api, ColumnType, KanbanType, SelectOptionType, SelectOptionsType, TableType, ViewType } from 'nocodb-sdk'
+import { useI18n } from 'vue-i18n'
+import { message } from 'ant-design-vue'
+import type { Row } from '~/composables/useViewData'
+import { SharedViewPasswordInj, deepCompare, enumColor, useNuxtApp } from '#imports'
+
+type GroupingFieldColOptionsType = SelectOptionType & { collapsed: boolean }
+
+export function useKanbanViewData(
```

Comment:
Instead of using `useState` for everything why not wrap this composable in `useInjectionState` (similar to other composables like this)
Reason:
* We execute the composable once and don't duplicate functions etc. 
* We don't create multiple injections for the same state (useState creates an injection for each value, i.e. all of these are passed through the whole tree instead of just 1 injection)